src/lib.rs

```diff
@@ -1481,3 +1482,30 @@ macro_rules! impl_tuple_zeroable {
 }
 
 impl_tuple_zeroable!(A, B, C, D, E, F, G, H, I, J);
+
+/// Trait for creating a [PinInitialized](PinInit) wrapper containing `T`.
+pub trait TryPinInitWrapper<T: ?Sized> {
+    /// Create an pin-initializer for a [`Self`] containing `T` form the `value` initializer.
+    fn try_pin_init<E>(value: impl PinInit<T, E>) -> impl PinInit<Self, E>;
```

Comment:
```suggestion
    fn pin_init<E>(value: impl PinInit<T, E>) -> impl PinInit<Self, E>;
```
Since it doesn't add any error of its own, I'd just name it this.

Reply:
Done